includes/class-wp-feature-query.php

```diff
@@ -35,20 +34,97 @@ public function __construct( $args = array() ) {
 				'type'       => array(),
 				'categories' => array(),
 				'location'   => array(),
+				'input_schema' => array(),
+				'output_schema' => array(),
 				'search'     => '',
 				'id'         => array(),
 			)
 		);
+	}
 
-		// Normalize array arguments.
-		foreach ( array( 'type', 'categories', 'location', 'id' ) as $key ) {
-			if ( ! is_array( $this->args[ $key ] ) ) {
-				$this->args[ $key ] = array( $this->args[ $key ] );
-			}
-
-			// Remove empty values.
-			$this->args[ $key ] = array_filter( $this->args[ $key ] );
-		}
+	/**
+	 * Get the schema for the query.
+	 *
+	 * @since 0.1.0
+	 * @return array The schema.
+	 */
+	public static function schema() {
+		return array(
+			'type' => array(
+				'description' => __( 'Filter features by their type.', 'features-api' ),
+				'type' => 'array',
+				'items' => array(
+					'type' => 'string',
+					'enum' => WP_Feature::TYPES,
+				),
+			),
+			'categories' => array(
+				'description' => __( 'Filter features by their categories.', 'features-api' ),
+				'type' => 'array',
+				'items' => array(
+					'type' => 'string',
+				),
+			),
+			'location' => array(
+				'description' => __( 'Filter features by their location (client/server).', 'features-api' ),
+				'type' => 'array',
+				'items' => array(
+					'type' => 'string',
+					'enum' => WP_Feature::LOCATIONS,
+				),
+			),
+			'input_schema' => array(
+				'description' => __( 'Filter features by their input schema.', 'features-api' ),
+				'type' => 'object',
+				'properties' => array(
+					'match' => array(
+						'type' => 'string',
+						'enum' => array( 'all', 'any' ),
+						'default' => 'any',
+					),
+					'fields' => array(
+						'type' => 'array',
+						'items' => array(
+							'type' => 'string',
+						),
+					),
+					'required' => array(
+						'fields',
+					),
+				),
+			),
+			'output_schema' => array(
+				'description' => __( 'Filter features by their output schema.', 'features-api' ),
+				'type' => 'object',
+				'properties' => array(
+					'match' => array(
+						'type' => 'string',
+						'enum' => array( 'all', 'any' ),
+						'default' => 'any',
+					),
+					'fields' => array(
+						'type' => 'array',
+						'items' => array(
+							'type' => 'string',
+						),
+					),
+					'required' => array(
+						'fields',
+					),
+				),
+			),
+			'search' => array(
+				'description' => __( 'Search features by name, description, or ID.', 'features-api' ),
+				'type' => 'string',
+			),
+			'id' => array(
```

Comment:
We should not include the search by id on the collection to be consistent with endpoints like posts (also does not support ID search). If someone has the ID the person can request that resource only instead of using the collection endpoint,

Reply:
good point